### DISCUSSION TITLE
One OL Layer per style-file layer?

### DISCUSSION THREAD

**Discussion by smellyshovel** [openlayers/ol-mapbox-style#1207] [Q&A] [answered]
Hello! First of all, thanks for this amazing tool, it's a real time saver.

I have a question though. Would it possible somehow to tell the `apply` function to create one separate OL Layer per one `layer` declaration in the provided style file?

Currently, if I have a style file with multiple layers declared, after calling the `apply` function with this style file, if I call the `map.getLayers()`, it returns me only 1 layer. As I understand, this single layer combines all the layers from the style file.

However, I'd really prefer it to create 1 OL Layer for each style-file-declared one. I need this to be able to toggle their visibility independently.

Is this somehow achievable? Thanks!

**Comment by ahocevar** [comment]
Layers in a Mapbox Style document are a completely different concept than layers in OpenLayers. If you want to toggle the visibility of a layer from the Mapbox Style document, you can use the [updateMapboxLayer()](https://github.com/openlayers/ol-mapbox-style?tab=readme-ov-file#functionsupdatemapboxlayermd) function, setting the `layout` property `visibility`.

**Comment by smellyshovel** [comment]
@ahocevar thanks for the quick response.

I tried 2 different approaches, one that you suggested, and another one is removing a layer altogether, instead of trying to hide it. None of them seem to produce any effect. Do I miss something?

```
      apply(map, props.styleFileUrl);
      setTimeout(() => {
        // One way
        updateMapboxLayer(map, {
          id: "building",
          type: "fill",
          source: "carto",
          "source-layer": "building",
          layout: {
            visibility: "none"
          }
        });

        // or another
        removeMapboxLayer(map, "water");
      }, 10000);
```

Here's the link to the style file, just in case: https://basemaps.cartocdn.com/gl/voyager-gl-style/style.json

**Comment by ahocevar** [comment, marked as answer]
For "building", you won't see much difference because "building_top" is on top of "building". And for "water", you won't see much difference either, because "water_shadow" is on top of it.

See https://codesandbox.io/p/sandbox/mapbox-style-forked-8psldc for an example that sets layers whose id starts with `"building"` or `"water"` to invisible.

**Reply by smellyshovel** [reply]
Got it, my bad. Thank you for your help!
